Adapt test script to easily test with another pipelines version

eddycharly opened the pull request:
<!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

# Changes

This PR adapts the test script to ease testing with another pipelines version.

It does so by allowing passing a version to the pipelines installer.
An end user can override the pipelines version deployed by defining the `PIPELINES_VERSION` environment variable:
```base
export PIPELINES_VERSION=v0.12.1
```

The default pipelines version used in our integration tests bumps from `v0.11.0` to `v0.13.2`.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

eddycharly commented:
/cc @a-roberts @AlanGreene

eddycharly commented:
We can reduce the list too, leaving `v0.11.0` should be as long as before.
But it's easier to control the version used by the tests.
By the way, shouldn't we test with pipelines `v0.13.2` now ?

AlanGreene commented:
Yes we should test with the latest 👍 

Maybe it would be useful to support running against multiple versions in the local-e2e-tests script, so we can still easily run this if we're making changes we expect to impact older versions, but don't have to carry that burden on every PR.

eddycharly commented:
Changed to test only `v0.13.2`.

a-roberts commented:
I like the idea of running this against multiple versions but agree doing it on every PR is too much - so let's add some docs to inform users how to do this and when they would want to (e.g. when testing against a new version). And we can mention how to add a new version to the table. I think this, potentially combined with the Postman tests (if we deem that adds sufficient new coverage) is a good step in the right direction

eddycharly commented:
I agree that running it on every PR is a bit overkill but at least now we can.

eddycharly commented:
TBH, i don't see what to put in docs, this is just the same as before but just changed the implementation a little bit to allow specifying a pipelines version from the caller when installing tekton pipelines

a-roberts commented:
> TBH, i don't see what to put in docs, this is just the same as before but just changed the implementation a little bit to allow specifying a pipelines version from the caller when installing tekton pipelines

Sure, so what I'm thinking is: the title of this commit/PR is "Test with multiple pipelines versions", and I think that's really useful.

It would be helpful for us, as we have to deal with "currency" e.g. newer pipeline versions.

You added `for pipelineVersion in v0.13.2` in one of the scripts. 

I'm thinking: repurpose this PR to allow for the testing of multiple pipeline versions, and then as a follow-up, which anyone could do, make it easy for developers to add to that for loop, for their own local testing. This would be useful, say, when Pipelines 0.14 is announced, and we want to run something that tells us at least where to look for potential problems - instead of doing everything manually. We could also run it periodically, and against any given previous version.

I remember working on a similar issue over a year ago and I had text file that was parsed - we could do something similar or just except an arg list.

a-roberts commented:
Lemme know if you plan on doing anything further @eddycharly, I would also amend your commit message in the same way you've done for the PR title, then happy to lgtm

eddycharly commented:
Changed the PR title and updated description, more inline with the content of the PR.

Just an idea, we could run several integration tests instances, one per pipelines version `pull-tekton-dashboard-integration-tests-pipelines-0-11-0`, `pull-tekton-dashboard-integration-tests-pipelines-0-12-1`, `pull-tekton-dashboard-integration-tests-pipelines-0-13-2`.

a-roberts commented:
> Changed the PR title and updated description, more inline with the content of the PR.
> 
> Just an idea, we could run several integration tests instances, one per pipelines version `pull-tekton-dashboard-integration-tests-pipelines-0-11-0`, `pull-tekton-dashboard-integration-tests-pipelines-0-12-1`, `pull-tekton-dashboard-integration-tests-pipelines-0-13-2`.



> Changed the PR title and updated description, more inline with the content of the PR.
> 
> Just an idea, we could run several integration tests instances, one per pipelines version `pull-tekton-dashboard-integration-tests-pipelines-0-11-0`, `pull-tekton-dashboard-integration-tests-pipelines-0-12-1`, `pull-tekton-dashboard-integration-tests-pipelines-0-13-2`.

Yeah, and we could do that at release time especially (not sure how we'd turn that on in an easy way - so we could have links to additional scripts in the Prow config that are only picked up on some flag), or on demand locally through a follow-up PR I discussed above.

Either way, let's get this in and thanks 👍 

/lgtm

eddycharly commented:
In the end i added a `PIPELINES_VERSION` env variable to allow overriding the version.

Description updated.

eddycharly commented:
In prow, we could add jobs, one per pipelines version, each job would have a different `PIPELINES_VERSION` env var... do you want that i give it a try ?

a-roberts commented:
> In prow, we could add jobs, one per pipelines version, each job would have a different `PIPELINES_VERSION` env var... do you want that i give it a try ?

Yep, worth it to see what would happen 😸

eddycharly commented:
Here we go https://github.com/tektoncd/plumbing/pull/432
